sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -212,9 +212,12 @@ static void CleanupThread()
           Logger::Level::Verbose,
           "Clean pool - no connections on wake - return *************************");
       CurlConnectionPool::g_curlConnectionPool.IsCleanThreadRunning = false;
-      return;
+      break;
```

Comment:
return and break should be the same here, IMO.  But I see you want to reach the normal function flow by just breaking the infinite loop.  Ok.